cobbler/items/item.py

```diff
@@ -990,14 +1220,59 @@ def grab_tree(self) -> list:
                  itself and the settings of Cobbler.
         """
         results = [self]
-        parent = self.parent
+        parent = self.logical_parent
         while parent is not None:
             results.append(parent)
-            parent = parent.parent
+            parent = parent.logical_parent
             # FIXME: Now get the object and check its existence
         results.append(self.api.settings())
         self.logger.debug(
             "grab_tree found %s children (including settings) of this object",
             len(results),
         )
         return results
+
+    @property
+    def cache(self) -> ItemCache:
+        """
+        Gettinging the ItemCache oject.
+
+        .. note:: This is a read only property.
+
+        :getter: This is the ItemCache oject.
+        """
+        return self._cache
+
+    def _clean_dict_cache(self, name: Optional[str]):
+        """
+        Clearing the Item dict cache.
+
+        :param obj: The object whose modification invalidates the dict cache.
+                    Can be Item, Settings or SIGNATURE_CACHE.
+        :param name: The name of Item attribute or None.
+        """
+        if not self.api.settings().cache_enabled:
+            return
+
+        if name is not None:
+            attr = getattr(type(self), name[1:])
+            if (
+                isinstance(attr, (InheritableProperty, InheritableDictProperty))
+                and self.api.get_items(self.COLLECTION_TYPE).get(self.name) is not None
+            ):
+                # Invalidating "resolved" caches
+                for dep_item in self.descendants:
+                    dep_item.cache.set_dict_cache(None, True)
+
+        # Invalidating the cache of the object itself.
+        self.cache.clean_dict_cache()
+
+    def clean_cache(self, name: Optional[str] = None):
+        """
+        Clearing the Item cache.
+
+        :param obj: The object whose modification invalidates the dict cache.
+                    Can be Item, Settings or SIGNATURE_CACHE.
+        :param name: The name of Item attribute or None.
+        """
+        self._clean_dict_cache(name)
```

Comment:
Any reason why those are two methods and not a single one?

Reply:
At the moment, two methods is redundant, but if we need to cache anything other than `to_dict`, we can write a new a new `clean_"something else"` method and add its call to `clean_cache`.

Reply:
Very well. That is more then reasonable. For a system this might a Network Interface in the future.